stdlib/http/src/main/java/org/ballerinalang/net/http/ClientHandshakeListener.java

```diff
@@ -0,0 +1,126 @@
+                HttpUtil.createResponseStruct(carbonResponse));
+        ObjectValue webSocketConnector = BallerinaValues.createObjectValue(PROTOCOL_PACKAGE_HTTP,
+                WebSocketConstants.WEBSOCKET_CONNECTOR);
+        WebSocketOpenConnectionInfo connectionInfo = new WebSocketOpenConnectionInfo(
+                wsService, webSocketConnection, webSocketClient);
+        webSocketConnector.addNativeData(WebSocketConstants.NATIVE_DATA_WEBSOCKET_CONNECTION_INFO, connectionInfo);
+        clientConnectorListener.setConnectionInfo(connectionInfo);
+        webSocketClient.set(WebSocketConstants.CLIENT_CONNECTOR_FIELD, webSocketConnector);
+        if (webSocketClient.getMapValue(CLIENT_ENDPOINT_CONFIG).getMapValue(RETRY_CONFIG) == null) {
+            if (readyOnConnect) {
+                webSocketConnection.readNextFrame();
+            }
+            WebSocketUtil.populateEndpoint(webSocketConnection, webSocketClient);
+            logger.info(CONNECTED_TO + webSocketClient.getStringValue(WebSocketConstants.
+                    CLIENT_URL_CONFIG));
+        } else {
+            RetryContext retryConfig = (RetryContext) webSocketClient.getNativeData(RETRY_CONFIG);
+            logger.info(CONNECTED_TO + webSocketClient.getStringValue(WebSocketConstants.
+                    CLIENT_URL_CONFIG));
+            if (retryConfig.getConnectionState()) {
+                webSocketClient.set(WebSocketConstants.LISTENER_ID_FIELD, webSocketConnection.getChannelId());
+            } else {
+                WebSocketUtil.populateEndpoint(webSocketConnection, webSocketClient);
+            }
+            if (retryConfig.getConnectionState() || readyOnConnect) {
+                webSocketConnection.readNextFrame();
+            }
+            retryConfig.setConnectionState();
+            retryConfig.setReconnectAttempts(0);
+        }
+        countDownLatch.countDown();
+    }
+
+    @Override
+    public void onError(Throwable throwable, HttpCarbonResponse response) {
+        if (response != null) {
+            webSocketClient.set(WebSocketConstants.CLIENT_RESPONSE_FIELD, HttpUtil.createResponseStruct(response));
+        }
+        ObjectValue webSocketConnector = BallerinaValues.createObjectValue(PROTOCOL_PACKAGE_HTTP,
+                WebSocketConstants.WEBSOCKET_CONNECTOR);
+        WebSocketOpenConnectionInfo connectionInfo = new WebSocketOpenConnectionInfo(
+                wsService, null, webSocketClient);
+        webSocketConnector.addNativeData(WebSocketConstants.NATIVE_DATA_WEBSOCKET_CONNECTION_INFO, connectionInfo);
+        countDownLatch.countDown();
+        if (throwable instanceof IOException || throwable instanceof BlockingOperationException) {
+            if (hasRetryConfig(webSocketClient)) {
+                if (!doReconnect(connectionInfo)) {
+                    WebSocketDispatcher.dispatchError(connectionInfo, throwable);
+                    logger.info(STATEMENT_FOR_RECONNECT +
```

Comment:
This log message is something related to failure in reconnecting, which IMO is more suitable if it is inside the doReconnect() method itself.